docs/pages/kubernetes-access/troubleshooting.mdx

```diff
@@ -0,0 +1,212 @@
+```
+
+### Symptoms
+
+After configuring a GKE Autopilot cluster in Teleport, all attempts to retrieve
+Kubernetes Objects fail with an error similar to:
+
+Or the following:
+
+```text
+GKE Autopilot denies requests that impersonate "system:masters" group
+```
+
+<Notice type="note">
+This issue only affects GKE Autopilot clusters. If you're using a standard GKE
+cluster, this issue doesn't apply to you.
+</Notice>
+
+### Explanation
+
+Unlike the standard Kubernetes cluster, Autopilot clusters forbid requests to
+impersonate system identities. This is a security feature that prevents users
+from gaining access to the cluster's control plane and performing administrative
+actions if they can impersonate users or groups.
+[Learn more about GKE Autopilot security](https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-security#built-in-security).
+
+Teleport uses impersonation to retrieve Kubernetes objects
+on behalf of the user. This is done by sending a request to the Kubernetes API server
+with the user's identity in the `Impersonate-User` header and all the Kubernetes
+Groups they can use in the `Impersonate-Group` header.
+
+Since `system:masters` is a built-in Kubernetes group in all clusters, it's usual
+for administrators to use it to gain administrative access to the cluster's control
+plane. However, Autopilot clusters forbid impersonating this group and require that
+another group is used instead.
+
+### Resolution
+
+Per the description above, the solution is to configure a different group for
+impersonation in Teleport. This can be done by setting the role's
+`kubernetes_group` parameter to a Group that the Autopilot cluster allows to
+impersonate.
+
+The `teleport-kube-agent` chart configures a Kubernetes Group with the same access
+levels as the `system:masters` group when it detects the target cluster is a GKE
+cluster.
+This group is named, by default, `cluster-admin`, but the name can be changed
+by setting the `adminClusterRoleBinding.name` parameter.
+
+The Kubernetes Group isn't created automatically when installing the chart on
+non-GKE clusters, so please don't change the `kubernetes_groups` parameter to
+`cluster-admin` unless you created the group manually or installed the chart with
+`adminClusterRoleBinding.create` parameter set to `true`.
+
+It's important to note that the group must be configured in the cluster before
+it can be used for impersonation. If the group is not configured, the impersonation
+request will fail with a `403 Forbidden` error and the user will not be able to
+access the cluster.
+
+When you opt to continue using the `system:masters` group for impersonation in
+non-Autopilot clusters, you must ensure that the Teleport roles that grant
+`system:masters` access can't be used to access GKE Autopilot clusters.
+
+As an example, a user with the following role can impersonate the `system:masters`
+group in any Kubernetes cluster:
+
+```yaml
+kind: role
+version: v5
```

Comment:
Should we make this a v6 role?

Reply:
Actually for master, it should be `v7` :laughing: